crates/sui/src/client_ptb/ast.rs

```diff
@@ -0,0 +1,353 @@
+    Publish(Spanned<String>),
+    Upgrade(Spanned<String>, Spanned<Argument>),
+    WarnShadows,
+    Preview,
+}
+
+/// An enum representing the parsed arguments of a PTB command.
+#[derive(Debug, Clone)]
+pub enum Argument {
+    Bool(bool),
+    U8(u8),
+    U16(u16),
+    U32(u32),
+    U64(u64),
+    U128(u128),
+    U256(move_core_types::u256::U256),
+    Gas,
+    Identifier(String),
+    VariableAccess(Spanned<String>, Vec<Spanned<String>>),
+    Address(NumericalAddress),
+    String(String),
+    Vector(Vec<Spanned<Argument>>),
+    Option(Spanned<Option<Box<Argument>>>),
+}
+
+impl Argument {
+    /// Resolve an `Argument` into a `MoveValue` if possible. Errors if the `Argument` is not
+    /// convertible to a `MoveValue`.
+    pub fn to_pure_move_value(&self, loc: Span) -> PTBResult<MoveValue> {
+        Ok(match self {
+            Argument::Bool(b) => MoveValue::Bool(*b),
+            Argument::U8(u) => MoveValue::U8(*u),
+            Argument::U16(u) => MoveValue::U16(*u),
+            Argument::U32(u) => MoveValue::U32(*u),
+            Argument::U64(u) => MoveValue::U64(*u),
+            Argument::U128(u) => MoveValue::U128(*u),
+            Argument::U256(u) => MoveValue::U256(*u),
+            Argument::Address(a) => MoveValue::Address(a.into_inner()),
+            Argument::Vector(vs) => MoveValue::Vector(
+                vs.iter()
+                    .map(|sp!(loc, v)| v.to_pure_move_value(*loc))
+                    .collect::<PTBResult<Vec<_>>>()
+                    .map_err(|e| e.with_help(
+                            format!("Was unable to parse '{self}' as a pure PTB value. This is most likely because \
+                                    the vector contains non-primitive (e.g., object or array) \
+                                    values which aren't permitted inside vectors")
+                            ))?
+            ),
+            Argument::String(s) => {
+                MoveValue::Vector(s.bytes().map(MoveValue::U8).collect::<Vec<_>>())
+            }
+            Argument::Option(sp!(loc, o)) => {
+                if let Some(v) = o {
+                    let v = v.as_ref().to_pure_move_value(*loc).map_err(|e| e.with_help(
+                            format!(
+                                "Was unable to parse '{self}' as a pure PTB value. This is most likely because \
+                                the option contains a non-primitive (e.g., object or array) \
+                                value which isn't permitted inside an option"
+                                )
+                            ))?;
+                    MoveValue::Vector(vec![v])
+                } else {
+                    MoveValue::Vector(vec![])
+                }
+            }
+            Argument::Identifier(_)
+            | Argument::VariableAccess(_, _)
+            | Argument::Gas => error!(loc, "Was unable to convert '{self}' to primitive value (i.e., non-object value)"),
```

Comment:
I think it should be fine without the help message since we are pointing to the `gas` argument, plus will in (I think all, or almost all) cases add the help message from above.  

E.g. 
```
❱ sui client ptb --move-call "std::option::is_none" "<u64>" vector[gas]  --gas-budget 100000000
Encountered error when building PTB:
  × Error when processing PTB
   ╭─[1:47]
 1 │ --move-call std::option::is_none <u64> vector[gas]
   ·                                               ─┬─
   ·                                                ╰── Unable to convert 'gas' to non-object value.
 2 │ --gas-budget 100000000
   ╰────
  help: Literal vectors cannot contain object values.
```

Reply:
Thanks for explaining and for the example.